src/types.ts

```diff
@@ -24,10 +24,20 @@ export type RatelimitResponse = {
    * Maximum number of requests allowed within a window.
    */
   limit: number;
+  /**
+   * Maximum number of custom tokens allowed within a window.
+   * @default 0
+   */
+  rateLimit?: number;
   /**
    * How many requests the user has left within the current window.
    */
   remaining: number;
+  /**
+   * How many custom tokens the user has left within the current window.
+   * @default 0
+   */
+  remainingRates?: number;
```

Comment:
The difference between `remaining` and `remainingRates` is not trivial, I think we should only have single object for this purpose, based on the logic of my comment on `fixedWindow` file

Reply:
Agreed